examples/leptos-trunk/flake.nix

```diff
@@ -0,0 +1,106 @@
+{
+  description = "leptos-trunk example";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    rust-flake.url = "github:juspay/rust-flake";
+  };
+
+  outputs = inputs:
+    inputs.flake-parts.lib.mkFlake { inherit inputs; } {
+      systems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
+      imports = [
+        inputs.rust-flake.flakeModules.default
+        inputs.rust-flake.flakeModules.nixpkgs
+      ];
+      perSystem = { pkgs, self', lib, config, ... }:
+        let
+          rustToolchainFor =
+            p:
+            p.rust-bin.stable.latest.default.override {
+              # Set the build targets supported by the toolchain,
+              # wasm32-unknown-unknown is required for trunk
+              targets = [ "wasm32-unknown-unknown" ];
+            };
+
+          craneLib = config.rust-project.crane-lib.overrideToolchain rustToolchainFor;
+          src = lib.cleanSourceWith {
+            src = ./.; # The original, unfiltered source
+            filter = path: type:
+              (lib.hasSuffix "\.html" path) ||
+              # Default filter from crane (allow .rs files)
+              (craneLib.filterCargoSources path type);
+          };
+
+          commonArgs = {
+            inherit src;
+            strictDeps = true;
+            # We must force the target, otherwise cargo will attempt to use your native target
+            CARGO_BUILD_TARGET = "wasm32-unknown-unknown";
+
+            buildInputs =
+              [
+                # Add additional build inputs here
+              ]
+              ++ lib.optionals pkgs.stdenv.isDarwin [
+                # Additional darwin specific inputs can be set here
+                pkgs.libiconv
+              ];
+          };
+
+          cargoArtifacts = craneLib.buildDepsOnly (
+            commonArgs
+            // {
+              # You cannot run cargo test on a wasm build
+              doCheck = false;
+            }
+          );
+          my-app = craneLib.buildTrunkPackage (
```

Comment:
So, this example doesn't usae `rust-flake` at all? If there's no straight forward way to use leptops **via rust-flake**, we should remove this example until the time we have one.

Reply:
There's no point to keeping examples like this rust-flake repo, of course.

Reply:
`craneLib` comes from `rust-flake`, so we are actually using `rust-flake`. My choice of the name `craneLib` might be a bit misleading; I can rename it to `rust-flake-lib` or `rustLib` instead, or do you have any suggestions?

Reply:
You can open issue for asking what rust-flake can implement in order to make working with Leptos projects seemless.